topi/python/topi/cuda/batch_matmul.py

```diff
@@ -51,55 +62,73 @@ def _schedule(op):
             C = s.outputs[0].output(0)
 
         b, y, x = s[C].op.axis
-        y_bn = get_max_power2_factor(M, 64)
-        x_bn = get_max_power2_factor(N, 64)
-        by, y = s[C].split(y, y_bn)
-        bx, x = s[C].split(x, x_bn)
-        y_nthreads = min(y_bn, 8)
-        x_nthreads = min(x_bn, 8)
-        ty, yi = s[C].split(y, nparts=y_nthreads)
-        tx, xi = s[C].split(x, nparts=x_nthreads)
-        thread_x = te.thread_axis((0, x_nthreads), "threadIdx.x")
-        thread_y = te.thread_axis((0, y_nthreads), "threadIdx.y")
+
+        cfg.define_split("tile_y", y, num_outputs=3)
+        cfg.define_split("tile_x", x, num_outputs=3)
+        cfg.define_knob("auto_unroll_max_step", [8, 16, 32, 64])
+        target = tvm.target.Target.current()
+        if target.target_name in ['nvptx', 'rocm']:
```

Comment:
It reminds me that you may also need to add batch_matmul op strategy to ROCM; otherwise ROCM will never use this schedule.

Reply:
See, one might easily think so. But I know that ROCm uses the schedule. My impression is that calling the shared schedules GPU would make that more clear.

Reply:
Yeah I totally agree. That's why I suggested having another RFC/PR to fix all cases. On the other hand, simply specify `gpu` in `topi/cuda` and cover both CUDA and ROCM are also confusion to me.

Reply:
Yeah, I'm sticking to cuda for now to not create two confusions where there was only one. :slightly_smiling_face: